src/libopensc/card-piv.c

```diff
@@ -39,14 +39,30 @@
 #endif
 
 #ifdef ENABLE_OPENSSL
-	/* openssl only needed for card administration */
+	/* openssl needed for card administration and Secure Messaging */
 #include <openssl/evp.h>
 #include <openssl/bio.h>
 #include <openssl/pem.h>
 #include <openssl/rand.h>
 #include <openssl/rsa.h>
+#include <openssl/sha.h>
+#if !defined(OPENSSL_NO_EC)
+#include <openssl/ec.h>
+#endif
+#include <openssl/cmac.h>
+#include <openssl/err.h>
+
+#define piv_log_openssl(C) \
+	do { if (C && C->debug > 0 && C->debug_file != 0) \
+		{ ERR_print_errors_fp(C->debug_file); } \
+	} while(0)
```

Comment:
If you decide to normalize this somehow (it would be probably for separate PR), I stumbled upon the similar implementation in westcos driver:
https://github.com/OpenSC/OpenSC/blob/38da7e1062d4eb3beefa50a2c4838b8ac4ae7066/src/libopensc/card-westcos.c#L58-L69

Reply:
This should be a separate PR, as catching OpenSSL errors in opensc-debug.log would be useful for all drivers.
For now I would like to not touch this.